### ISSUE TITLE
dropout masking

### ISSUE THREAD

**Issue by yiqiaoc11** [pytorch/pytorch#7808]
If you have a question or would like help and support, please ask at our
[forums](https://discuss.pytorch.org/).

If you are submitting a feature request, please preface the title with [feature request].
If you are submitting a bug report, please fill in the following details.

## Issue description
We found memory leaking with Dropout within one epoch. It recovers to be normal after deleting self.dp() ... It seems graph keeps growing. I assume torch/nn/_functions/dropout.py should have mask with requires_grad = False.

+-------------------------------+----------------------+----------------------+
| 28%   51C    P2    64W / 250W |   6854MiB / 11178MiB |      3%      Default |
+-------------------------------+----------------------+----------------------+
...
+-------------------------------+----------------------+----------------------+
| 29%   51C    P2    66W / 250W |   8986MiB / 11178MiB |     10%      Default |
+-------------------------------+----------------------+----------------------+
+-------------------------------+----------------------+----------------------+
| 29%   51C    P0    63W / 250W |  10152MiB / 11178MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+

## Code example

        self.dp = nn.Dropout(dropout)
...
        embed_p = self.x_proj(embed_p)  # embed_p ~ L * B * hidden_size
        embed_h = self.x_proj(embed_h)

        embed_p = self.dp(embed_p)
        embed_h = self.dp(embed_h) 

## System Info
Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

- PyTorch or Caffe2:
- How you installed PyTorch (conda, pip, source):
- Build command you used (if compiling from source):
- OS:
- PyTorch version:
- Python version:
- CUDA/cuDNN version:
- GPU models and configuration:
- GCC version (if compiling from source):
- CMake version:
- Versions of any other relevant libraries:

**Comment by yiqiaoc11** [comment]
I'm using 0.4.0 of pytorch.

**Comment by wengong-jin** [comment]
I found the same memory leak issue with my project. After removing dropout, the memory stays the same. When training with dropout, the memory keeps growing and reaches OOM eventually.

**Comment by ezyang** [comment]
Hi @wengong-jin, can you easily try running your project on pytorch v1.0.0? That would be helpful to know if it still leaks.

**Comment by wengong-jin** [comment]
@ezyang Thanks for your suggestion. Switching to v1.0.1 solved the leaking issue (not sure about v1.0.0 though).
